libs/api-client/src/Network/Wire/Client/Monad.hs

```diff
@@ -40,13 +50,23 @@ data Server = Server
 class (MonadHttp m, MonadLogger m, MonadIO m) => MonadClient m where
     getServer :: m Server
     getLogger :: m Logger
+    -- | Allows running operations in Client monad which has MonadUnliftIO which is invalid
+    -- to implement on 'Session'
+    liftClient :: Client a -> m a
```

Comment:
This was one of the annoying (but harmless) things I needed to add to get around some problems I had with unliftio. `Session` can't implement `UnliftIO` because it has `StateT`; so instead we allow some method of running the strictly less powerful `Client` monad within `m`. This is basically `liftIO` but for a different Monad type. It solves the issues with unliftio; and every `MonadClient` can easily implement it anyways so it's not a big deal.